### ISSUE TITLE
Weighted/masked loss function for forces

### ISSUE THREAD

**Issue by messerlyra** [lanl/hippynn#152]
The current documentation for weighted/masked loss functions discusses training to energies and atom-valued targets (charges are mentioned). However, there is no clear documentation for training to forces with a weighted/masked loss function.

After some trial and error and guidance from Nicholas Lubbers, the following seems to have worked for a script based off the ani1x_training example:

In make_model:

force1w8_node = inputs.InputNode(db_name="force1_weights", index_state=hippynn.graphs.IdxType.MolAtom)
energy1w8_node = inputs.InputNode(db_name="energy1_weights", index_state=hippynn.graphs.IdxType.Molecules)

from hippynn.graphs.nodes.indexers import PaddingIndexer, AtomReIndexer, find_unique_relative

padding = find_unique_relative(network, PaddingIndexer)
real = padding.real_atoms
force1w8_node = AtomReIndexer("force1_weights", (force1w8_node, real))
force1w8_node.db_name = "force1_weights"

In make_loss:

"T-wRMSE": WeightedMSELoss.of_node(henergy1,energy1w8_node)**(1/2)
"F-wRMSE": WeightedMSELoss.of_node(force1,force1w8_node)**(1/2)

In load_db:

  n_samples = database.arr_dict['atomic_numbers'].shape[0]
  energy_shape = database.arr_dict[en_name1].shape
  force_shape = database.arr_dict[force_name1].shape
  energy1_mask = generate_boolean_mask(n_samples, energy_shape, pct_true=args.energy_percent1/100.).astype(np.float32)
  force1_mask = np.broadcast_to(energy1_mask[:, None, None], force_shape) ###In my use case, I want to mask energies and forces for the same snapshot, i.e., if I am masking the energy for a snapshot I do not want any of the forces from that snapshot
  database.arr_dict['energy1_weights'] = energy1_mask
  database.arr_dict['force1_weights'] = force1_mask

Auxiliary function to create mask:

def generate_boolean_mask(n_samples: int, shape: tuple, pct_true: float = 1.0):
    total_elements = np.prod(shape)
    k = int(total_elements * pct_true)
    mask = np.array([True] * k + [False] * (total_elements - k))
    np.random.shuffle(mask)
    return mask.reshape(shape)

Is there a simpler way to run weighted/masked loss functions for forces?

**Comment by lubbersnick** [comment]
Fixed with #156. Please see also that now you can create the loss function value using the db_name directly instead of building an input node explicitly.
